crates/wasmparser/src/validator/component_types.rs

```diff
@@ -3094,16 +3090,16 @@ impl<'a> SubtypeCx<'a> {
         }
     }
 
-    fn core_func_type(&self, a: &FuncType, b: &FuncType, offset: usize) -> Result<()> {
+    fn core_func_type(&self, a: CoreTypeId, b: CoreTypeId, offset: usize) -> Result<()> {
         if a == b {
```

Comment:
Could this include a debug assert that both types are in the same original shared tail of the type list?

Reply:
Specifically in the case that they are equal? Because if they are not equal, that is not guaranteed to be true.